File: VB/CS/E911.Module/Controllers/ShowCustomFormWindowController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports System.Collections.Generic
Imports DevExpress.ExpressApp.SystemModule

Namespace E911.Module.Controllers
	Public MustInherit Class ShowCustomFormWindowController
		Inherits WindowController
		Private navigationController As ShowNavigationItemController
		Public Sub New()
			TargetWindowType = WindowType.Main
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			navigationController = Frame.GetController(Of ShowNavigationItemController)()
			If navigationController IsNot Nothing Then
				AddHandler navigationController.CustomShowNavigationItem, AddressOf navigationController_CustomShowNavigationItem
			End If
		End Sub
		Protected Overrides Sub OnDeactivated()
			If navigationController IsNot Nothing Then
				RemoveHandler navigationController.CustomShowNavigationItem, AddressOf navigationController_CustomShowNavigationItem
			End If
			MyBase.OnDeactivated()
		End Sub
		Private Sub navigationController_CustomShowNavigationItem(ByVal sender As Object, ByVal e As CustomShowNavigationItemEventArgs)
			If e.ActionArguments.SelectedChoiceActionItem.Id = "CustomForm" Then
				ShowCustomForm()
				e.Handled = True
			End If
		End Sub
		Protected MustOverride Sub ShowCustomForm()
	End Class
End Namespace

File: CS/CS/E911.Module/Controllers/ShowCustomFormWindowController.cs
using System;
using DevExpress.ExpressApp;
using System.Collections.Generic;
using DevExpress.ExpressApp.SystemModule;

namespace E911.Module.Controllers {
    public abstract class ShowCustomFormWindowController : WindowController {
        private ShowNavigationItemController navigationController;
        public ShowCustomFormWindowController() {
            TargetWindowType = WindowType.Main;
        }
        protected override void OnActivated() {
            base.OnActivated();
            navigationController = Frame.GetController<ShowNavigationItemController>();
            if (navigationController != null)
                navigationController.CustomShowNavigationItem += navigationController_CustomShowNavigationItem;
        }
        protected override void OnDeactivated() {
            if (navigationController != null)
                navigationController.CustomShowNavigationItem -= navigationController_CustomShowNavigationItem;
            base.OnDeactivated();
        }
        private void navigationController_CustomShowNavigationItem(object sender, CustomShowNavigationItemEventArgs e) {
            if (e.ActionArguments.SelectedChoiceActionItem.Id == "CustomForm") {
                ShowCustomForm();
                e.Handled = true;
            }
        }
        protected abstract void ShowCustomForm();
    }
}
